t/ui/27-plugin_obs_rsync_status_details.t

```diff
@@ -0,0 +1,188 @@
+use Mojo::Server::Daemon;
+use Mojo::IOLoop::Server;
+use Mojo::IOLoop::ReadWriteProcess 'process';
+use Mojo::IOLoop::ReadWriteProcess::Session 'session';
+use Time::HiRes 'sleep';
+
+my $test_case   = OpenQA::Test::Case->new;
+my $schema_name = OpenQA::Test::Database->generate_schema_name;
+my $schema      = $test_case->init_data(schema_name => $schema_name);
+
+$SIG{INT} = sub {
+    session->clean;
+};
+
+END { session->clean }
+
+my $port = Mojo::IOLoop::Server->generate_port;
+my $host = "http://127.0.0.1:$port";
+
+sub fake_api_server {
+    my $mock = Mojolicious->new;
+    $mock->mode('test');
+    $mock->routes->get(
+        '/public/build/:proj/_result' => sub {
+            my $c    = shift;
+            my $proj = $c->render('proj');
+            $c->render(
+                status => 200,
+                text   => '<result project="'
+                  . $proj
+                  . '" repository="images" arch="local" code="published" state="published">'
+            );
+        });
+    return $mock;
+}
+
+sub _port { IO::Socket::INET->new(PeerAddr => '127.0.0.1', PeerPort => shift) }
+
+my $daemon;
+my $mock            = Mojolicious->new;
+my $server_instance = process sub {
+    $daemon = Mojo::Server::Daemon->new(app => fake_api_server, listen => [$host]);
+    $daemon->run;
+    _exit(0);
+};
+
+sub start_server {
+    $server_instance->set_pipes(0)->start;
+    sleep 0.1 while !_port($port);
+    return;
+}
+
+sub stop_server {
+    # now kill the worker
+    $server_instance->stop();
+}
+
+my $url = "http://127.0.0.1:$port/public/build/%%PROJECT/_result?package=000product";
+
+$ENV{OPENQA_CONFIG} = my $tempdir = tempdir;
+my $home_template = path(__FILE__)->dirname->dirname->child('data', 'openqa-trigger-from-obs');
+my $home          = "$tempdir/openqa-trigger-from-obs";
+
+$tempdir->child('openqa.ini')->spurt(<<"EOF");
+[global]
+plugins=ObsRsync
+[obs_rsync]
+home=$home
+project_status_url=$url
+EOF
+
+note("Starting fake api server");
+start_server();
+
+my $driver = call_driver(sub { }, {with_gru => 1});
+
+plan skip_all => $OpenQA::SeleniumTest::drivermissing unless $driver;
+
+$driver->find_element_by_class('navbar-brand')->click();
+$driver->find_element_by_link_text('Login')->click();
+
+my %params = (
+    'Proj1'       => ['190703_143010'],
+    'BatchedProj' => ['191216_150610'],
+    'Batch1'      => ['191216_150610', 'BatchedProj'],
+);
+
+sub _wait_helper {
+    my ($element, $test_break) = @_;
+    my $ret;
+    for (my $retries = 0; $retries < 50; $retries = $retries + 1) {
+        $ret = $driver->find_element($element)->get_text();
+        last if &$test_break($ret);
+        sleep(0.1);
+    }
+    return $ret;
+}
+
+foreach my $proj (sort { $b cmp $a } keys %params) {
+    dircopy($home_template, $home);
+    my ($dt, $parent) = @{$params{$proj}};
+    $parent = "" unless $parent;
+
+    $driver->get("/admin/obs_rsync/$parent");
+    my $projfull = $proj;
+    $projfull = "$parent|$proj" if $parent;
```

Comment:
The comments and/or test descriptions in this block could be better. I don't really have any idea what's going on reading it.

Reply:
I don't think how additional noise can help here to those who haven't read existing documentation related to plugin, haven't invested time to understand its challenges, haven't tried existing UI or at least have idea what test did before the change.

Reply:
What have "challenges" to do with this? I just would like to know what is actually being tested, in case i ever have to work on this if there is an emergency.

Reply:
If there is an emergency you better understand what challenges are. To understand what is tested in UI test you must at least look at UI once. That said - I do not want to use review board as a chat service. If you see real issues - please log them. If you want to discuss something - there are better ways.

Reply:
Of course one can even just read the code itself to find out what it does but I find comments beneficial as well - even if they just summarize what a certain block does. You also wouldn't write a longer document like a bachelors thesis without headings although the headings don't provide any additional information.